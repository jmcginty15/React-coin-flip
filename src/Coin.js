import './Coin.css';

const Coin = ({ result }) => {
    const headsImage = 'https://www.usmint.gov/wordpress/wp-content/uploads/2019/11/2020-america-the-beautiful-quarters-coin-uncirculated-obverse-philadelphia-768x768.jpg';
    const tailsImage = 'https://www.usmint.gov/wordpress/wp-content/uploads/2019/11/2020-america-the-beautiful-quarters-coin-tallgrass-prairie-kansas-uncirculated-reverse-768x768.jpg';

    return (
        <div className="Coin">
            <img className="Coin-image" src={result === 'heads' ? headsImage : tailsImage} alt={result === 'heads' ? 'heads' : 'tails'} />
        </div>
    )
}

export default Coin;