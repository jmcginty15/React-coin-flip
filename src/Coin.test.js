import React from "react";
import { render } from "@testing-library/react";
import Coin from "./Coin";

// smoke test
it("renders without crashing", function () {
    render(<Coin result="heads" />);
    render(<Coin result="tails" />);
});

// snapshot test
it("matches heads snapshot", function () {
    const { asFragment } = render(<Coin result="heads" />);
    expect(asFragment()).toMatchSnapshot();
});

// snapshot test
it("matches tails snapshot", function () {
    const { asFragment } = render(<Coin result="tails" />);
    expect(asFragment()).toMatchSnapshot();
});
