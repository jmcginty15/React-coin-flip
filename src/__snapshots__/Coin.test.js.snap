// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches heads snapshot 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    <img
      alt="heads"
      class="Coin-image"
      src="https://www.usmint.gov/wordpress/wp-content/uploads/2019/11/2020-america-the-beautiful-quarters-coin-uncirculated-obverse-philadelphia-768x768.jpg"
    />
  </div>
</DocumentFragment>
`;

exports[`matches tails snapshot 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    <img
      alt="tails"
      class="Coin-image"
      src="https://www.usmint.gov/wordpress/wp-content/uploads/2019/11/2020-america-the-beautiful-quarters-coin-tallgrass-prairie-kansas-uncirculated-reverse-768x768.jpg"
    />
  </div>
</DocumentFragment>
`;
